pkg/beacon/beacon.go

```diff
@@ -25,32 +23,36 @@ var logger = log.Logger("keep-beacon")
 // internal random beacon implementation. Returns an error if this failed,
 // otherwise enters a blocked loop.
 func Initialize(
-	ctx context.Context,
 	operatorPublicKey *operator.PublicKey,
-	relayChain relaychain.Interface,
-	blockCounter chain.BlockCounter,
-	signing chain.Signing,
+	beaconChain beaconchain.Interface,
 	netProvider net.Provider,
 	persistence persistence.Handle,
 ) error {
-	chainConfig := relayChain.GetConfig()
+	chainConfig := beaconChain.GetConfig()
 
-	groupRegistry := registry.NewGroupRegistry(relayChain, persistence)
+	blockCounter, err := beaconChain.BlockCounter()
+	if err != nil {
+		return fmt.Errorf("failed to get block counter instance: [%v]", err)
+	}
+
+	signing := beaconChain.Signing()
+
+	groupRegistry := registry.NewGroupRegistry(beaconChain, persistence)
 	groupRegistry.LoadExistingGroups()
 
-	node := relay.NewNode(
+	node := newNode(
 		operatorPublicKey,
 		netProvider,
 		blockCounter,
 		chainConfig,
 		groupRegistry,
 	)
 
-	eventDeduplicator := event.NewDeduplicator(relayChain)
+	eventDeduplicator := event.NewDeduplicator(beaconChain)
 
-	node.ResumeSigningIfEligible(relayChain, signing)
+	node.ResumeSigningIfEligible(beaconChain, signing)
```

Comment:
`beaconChain` has `signing` inside, so we could pass just one parameter to this function.

Reply:
Yep, we will address that once we work on the relay entry for v2.